docs/macos-core-installation.md

```diff
@@ -72,17 +68,17 @@ cmake ../ \
 -DOPENSSL_CRYPTO_LIBRARIES=/usr/local/opt/openssl/lib/libcrypto.dylib
 ```
 
-To know the amount of cores available.
-You can use the following command
+To know the amount of threads available.
+You can use the following command:
 
 ```sh
 nproc --all
 ```
 
-Then, replacing `4` with the number of threads that you want to execute, type:
+Then you can simply run this to compile with all threads available and install the server.
 
 ```sh
-make -j 4
+make -j$(nproc --all)
```

Comment:
as far as I remember on macos the command is different than Linux (nproc is only available on Linux)

However, on our compiler script we use this: https://github.com/azerothcore/azerothcore-wotlk/blob/master/apps/compiler/includes/functions.sh#L81

But apparently on MacOS there's another command: https://github.com/memkind/memkind/issues/33#issuecomment-316615308

Maybe it's worth to check how to make it cross-platform and also update the compiler script

Reply:
this could use the `getconf` command which works on osx and linux, and yes nproc is available on osx trough homebrew.